File: internal/models/org.go
// agency-gateway/internal/models/org.go
package models

// OrgConfig is the schema for org.yaml.
type OrgConfig struct {
	Version        string `yaml:"version" validate:"required" default:"0.1"`
	Name           string `yaml:"name" validate:"required"`
	Operator       string `yaml:"operator" validate:"required"`
	Created        string `yaml:"created" validate:"required"`
	DeploymentMode string `yaml:"deployment_mode" validate:"required,oneof=standalone team enterprise" default:"standalone"`
}

File: internal/images/contexts/_shared/models/org.py
"""Organization configuration schema."""

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class OrgConfig(BaseModel):
    """Schema for org.yaml."""

    model_config = ConfigDict(extra="forbid")

    version: str = "0.1"
    name: str
    operator: str
    created: str
    deployment_mode: Literal["standalone", "team", "enterprise"] = "standalone"
